Out of time today (and I haven't looked at CentOS 7 yet), but here's a log from a modified xrdp instance tripping the filter:-

```
[2023-06-07T12:16:31.601+0100] [ERROR] [process_pending_sigchld_events(xrdp_listen.c:872)] Child 35041 was terminated with seccomp violation
```

then, feeding the PID into `journalctl`:-

```
$ sudo journalctl --no-pager  _AUDIT_TYPE_NAME=SECCOMP _PID=35041 -S today
Jun 07 12:16:31 xrdp-test audit[35041]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=unconfined pid=35041 comm="xrdp" exe="/home/mjb/xrdp/xrdp/.libs/xrdp" sig=31 arch=c000003e syscall=28 compat=0 ip=0x7f7fab91ed2b code=0x80000000
```

which gives the syscall of 28 (madvise)

I'll post my patch later.

--- a/instfiles/xrdp.service.in
+++ b/instfiles/xrdp.service.in
@@ -9,6 +9,8 @@ Type=exec
 EnvironmentFile=-@sysconfdir@/sysconfig/xrdp
 EnvironmentFile=-@sysconfdir@/default/xrdp
 ExecStart=@sbindir@/xrdp $XRDP_OPTIONS --nodaemon
+SystemCallArchitectures=native
+SystemCallFilter=@basic-io @file-system @io-event @ipc @network-io @process @signal ioctl madvise sysinfo uname
 
 [Install]
 WantedBy=multi-user.target

